src/installer/corehost/cli/apphost/bundle/header.cpp

```diff
@@ -9,18 +9,24 @@
 
 using namespace bundle;
 
-bool header_fixed_t::is_valid() const
+bool header_fixed_t::is_valid(bool exact_match) const
 {
+    // The AppHost expects the bundle_header to be an exact_match for which it was built.
+    // The framework accepts backwards compatible header versions.
+
     return num_embedded_files > 0 &&
-           ((major_version < header_t::major_version) ||
-            (major_version == header_t::major_version && minor_version <= header_t::minor_version));
+          (exact_match) ?
+              (major_version == header_t::major_version) && (minor_version == header_t::minor_version) :
+              ((major_version < header_t::major_version) ||
+                   (major_version == header_t::major_version && minor_version <= header_t::minor_version));
+
 }
```

Comment:
Can you write this in multiple `if` statements with an early return style? This is starting to get difficult to read.

Reply:
Also, please check if the version in the header is either 1 or 2 before any other test, and return `false` otherwise. Just a sanity check before making any assumptions.

Reply:
I've rewritten the check using `if() {} else {}` blocks.

> Also, please check if the version in the header is either 1 or 2 before any other test

I think this check is implicit, because `major_version` is a uint, that's bounded by `header.major_version` -- the constant `2`.